File: 151003/Baranouski/RV_Project/src/main/java/by/bsuir/RV_Project/dto/responses/converters/StoryResponseConverter.java
package by.bsuir.RV_Project.dto.responses.converters;

import by.bsuir.RV_Project.domain.Story;
import by.bsuir.RV_Project.dto.responses.StoryResponseDto;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface StoryResponseConverter {
    @Mapping(source = "author.id", target = "authorId")
    StoryResponseDto toDto(Story story);
}

File: 151003/Yagnnish/Lab1/Dtos/ResponseDtos/ResponseConverters/StoryResponseConverter.cs
using lab_1.Domain;

namespace lab_1.Dtos.ResponseDtos.ResponseConverters
{
    public class StoryResponseConverter : BaseResponse<StoryResponseDto, Story>
    {
        public StoryResponseDto ToDto(Story entity)
        {
            var dto = new StoryResponseDto();
            dto.content = entity.Content;
            dto.title = entity.Title;
            dto.created = entity.Created;
            dto.authorId = entity.AuthorId;
            dto.id = entity.Id;
            dto.modified = entity.Modified;
            return dto;
        }
    }
}
